code/modules/mining/lavaland/megafauna_loot.dm

```diff
@@ -1115,3 +1115,162 @@
 	playsound(target, 'sound/effects/magic/lightningbolt.ogg', 100, TRUE)
 	target.visible_message(span_danger("A thunderbolt strikes [target]!"))
 	explosion(target, light_impact_range = (boosted ? 1 : 0), flame_range = (boosted ? 2 : 1), silent = TRUE)
+
+
+/datum/action/innate/brain_undeployment
+	name = "Disconnect from shell"
+	desc = "Stop controlling your shell and resume normal core operations."
+	button_icon = 'icons/mob/actions/actions_AI.dmi'
+	button_icon_state = "ai_core"
+
+/datum/action/innate/brain_undeployment/Trigger(trigger_flags)
+	if(!..())
+		return FALSE
+	var/obj/item/organ/brain/cybernetic/ai/shell_to_disconnect = owner.get_organ_by_type(/obj/item/organ/brain/cybernetic/ai)
+
+	shell_to_disconnect.undeploy()
+	return TRUE
+
+/obj/item/organ/brain/cybernetic/ai
+	name = "AI-uplink brain"
+	desc = "Can be inserted into a body with NO ORGANIC ORGANS (robotics only) to allow AIs to control it. Comes with its own health sensors beacon. MUST be a humanoid or bad things happen to the consciousness."
+	can_smoothen_out = FALSE
+	/// if connected, our AI
+	var/mob/living/silicon/ai/mainframe
+	/// action for undeployment
+	var/datum/action/innate/brain_undeployment/undeployment_action = new
+
+/obj/item/organ/brain/cybernetic/ai/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/noticable_organ, "%PRONOUN_Their eyes move with machine precision, their expression completely blank.")
+
+/obj/item/organ/brain/cybernetic/ai/Destroy()
+	. = ..()
+	undeploy()
+	mainframe = null
```

Comment:
```suggestion
	undeploy()
```
seems undeploy() already nulls the mainframe

Reply:
if the brain was somehow deleted this will attempt to undeploy the AI but can fail if there was no mind in the first place somehow